### ISSUE TITLE
Performance drop when updating to latest Vcpkg version 

### ISSUE THREAD

**Issue by yknapp** [OSGeo/gdal#5579]
## Expected behavior and actual behavior.

I implemented searching a street within a given city based on a Germany OpenStreetMap file and GDAL.
Therefore, I use the street name as attribute filter and the city's envelope as spatial filter and loop through the results with `OGRLayer::getNextFeature`.  

For library management, I use [VCPKG](https://github.com/microsoft/vcpkg).
Firstly, I tested the code with `GDAL 3.4.1` and the database query was fast, as expected.
Then I updated VCPKG, so that GDAL was upated to `3.4.2`, and now the execution time of `getNextFeature` has increased a hundredfold (**from 11ms to 268ms**).
I'm not sure whether the problem is with GDAL or some of its dependencies.

## Steps to reproduce the problem.

I created a test project here: [https://github.com/yknapp/GdalTest/](https://github.com/yknapp/GdalTest/). It shows the different performances on the different GDAL versions. As per description in the readme I also provide two different VCPKG exports and the OpenStreetMap file.

## Operating system

Windows 10 x64
Compiler MSVC 2019 latest

## GDAL version and provenance

GDAL 3.4.1 and GDAL 3.4.2

**Comment by dg0yt** [comment]
FTR between GDAL 3.4.1 and 3.4.2, vcpkg also updated PROJ from 8.2.1 to 9.0.0.

**Comment by rouault** [comment]
> within a given city based on a Germany OpenStreetMap file

in which format exactly ? Can you point to such a file ?

**Comment by jratike80** [comment]
I am downloading the file (there is a link). By the name that contains "spatialite_osm_map" I guess it is a Spatialite db created with the OSM tools of Spatialite. I would like to see SQL that makes about the same thing as the code. Some others would probably like to see the code.

**Comment by yknapp** [comment]
> > within a given city based on a Germany OpenStreetMap file
> 
> in which format exactly ? Can you point to such a file ?

I created a simple project which should make it more clear: [https://github.com/yknapp/GdalTest/](https://github.com/yknapp/GdalTest/).  
The OpenStreetMap file is a spatialite database and was created with [spatialite-tools](https://www.gaia-gis.it/fossil/spatialite-tools/index).  
The OSM file can be found here: [https://drive.google.com/drive/u/1/folders/11TYCflrL2rzybpejnd9Gs7V3zjrGdvon](https://drive.google.com/drive/u/1/folders/11TYCflrL2rzybpejnd9Gs7V3zjrGdvon).

**Comment by jratike80** [comment]
The code seems to create a query like
`select * from ln_highway where NAME like 'Markt%'`

**Comment by yknapp** [comment]
@jratike80 

> The code seems to create a query like `select * from ln_highway where NAME like 'Markt%'`
Exactly, but I additionally I filter the results with a spatial filter, so I only get streets that are within the bounding box of the given city.

The SQL query would look like this:  
``` SQL
SELECT pkid, xmin, ymin, name
FROM idx_ln_highway_Geometry INNER JOIN ln_highway
ON idx_ln_highway_Geometry.pkid=ln_highway.id
WHERE (xmin < city_xmax AND ymin < city_ymax AND xmax > city_xmin AND ymax > city_y_min) AND name LIKE "Markt%";
```
`city_xmax` ... are the minimum and maximum x and y values of the city's bounding box.

**Comment by jratike80** [comment]
But the core issue seems to be why getNextFeature is slower. Do you really need a 19 GB Spatialite db for testing that?

**Comment by yknapp** [comment]
@jratike80
the target project must run on the 19GB db.

**Comment by jratike80** [comment]
Test with ogrinfo on GDAL 3.5.0dev is fast
`ogrinfo -sql "select * from ln_highway where NAME like 'Markt%'" germany-latest.sqlite -spat 11 47 12 48`

**Comment by kelteseth** [comment]
> Test with ogrinfo on GDAL 3.5.0dev is fast

Can you test it with GDAL 3.4.1? I think these tests do not directly relate to this problem, because It could be that there are more relate to the underlying dependencies that are used via vcpkg. Could you try to run the c++ test project (https://github.com/yknapp/GdalTest/)?

**Comment by dg0yt** [comment]
I would have recommended to use vcpkg manifest mode to pin all relevant dependencies and only vary gdal. Vcpkg also had sqlite3 changes. Maybe some of the new sqlite3 "features" must be explicitly turned on now to see original performance.
(And it goes without saying that we must not compare debug build with release build.)

**Comment by jratike80** [comment]
GDAL 3.4.1 seems to be equally fast. I used this 
`ogrinfo -sql "SELECT pkid, xmin, ymin, name FROM idx_ln_highway_Geometry INNER JOIN ln_highway ON idx_ln_highway_Geometry.pkid=ln_highway.id WHERE (xmin < 12 AND ymin < 48 AND xmax > 11 AND ymax > 47) AND name LIKE 'Markt%'" germany-latest.sqlite`

I can see that you use double quotes around a string here `AND name LIKE "Markt%";` and even is works with SQLite as long as there is no table with the same name you will get troubles some day. Use single quotes instead.
Use single quotes around strings even SQLite allows

**Comment by yknapp** [comment]
@jratike80 
> I apologize that I must use just a layman logic because I cannot program myself.

I didn't look into the GDAL source code, but I would expect exactly the same thing you wrote.

> GDAL 3.4.1 seems to be equally fast.

Thanks for testing. GDAL 3.4.1 was also very fast in my case, but GDAL 3.4.2 not.

**Comment by rouault** [comment]
I see that your GDAL 3.4.2 test case was with sqlite 3.37.2, while the 3.4.0 was with sqlite 3.37.0. However nothing in SQLite release notes indicates changes related to join / rtree, and when trying (on Linux) those versions, I can't see any perf difference. There's no change between GDAL 3.4.1 and 3.4.2 I can identify as being significant for this use case.
This works also fine with head of the 3.4 branch or GDAL master.

```
$ gdalinfo --version
GDAL 3.4.3, released 2022/04/22

$ time LD_LIBRARY_PATH=$HOME/install-sqlite-3.37.0/lib:$LD_LIBRARY_PATH ogrinfo ../../gdal/build_cmake/germany-latest.spatialite_osm_map  -spat 9.381513 47.637805 9.560811 47.737873 ln_highway -where "NAME like 'Markt%'" -al -q > /dev/null
real	0m0,151s

$ time LD_LIBRARY_PATH=$HOME/install-sqlite-3.37.2/lib:$LD_LIBRARY_PATH ogrinfo ../../gdal/build_cmake/germany-latest.spatialite_osm_map  -spat 9.381513 47.637805 9.560811 47.737873 ln_highway -where "NAME like 'Markt%'" -al -q > /dev/null
real	0m0,150s
```

So basically this is a "I can't reproduce" issue for me. Closing as such
